lib/internal/process/esm_loader.js

```diff
@@ -85,21 +84,21 @@ function loadModulesInIsolation(specifiers, loaders = []) {
   // A separate loader instance is necessary to avoid cross-contamination
   // between internal Node.js and userland. For example, a module with internal
   // state (such as a counter) should be independent.
-  const internalEsmLoader = new ESMLoader();
-  internalEsmLoader.addCustomLoaders(loaders);
+  const internalESMLoader = new ESMLoader();
```

Comment:
should `--loader` and `--import` run on the main thread?

Reply:
Moving `--loader` off the main thread is the point of this PR, so I’m not sure I understand the question?

`--import` is user application code so it needs to stay on the main thread.

Reply:
I think Moshe is asking whether both "internal" and "public" loaders should be in the worker thread (at the moment in this PR, only the public is—once I get the public, I'll move internal).

Reply:
If we move the “internal” loader onto the worker thread, wouldn’t that break user code that monkey-patches Node stuff?

Perhaps we could do that as a follow-up, if we do it at all. It feels like it might be a semver-major change.